packages/babel-cli/src/babel/util.js

```diff
@@ -101,6 +101,14 @@ export function deleteDir(path) {
   }
 }
 
+export function deleteFile(path) {
+  if (fs.existsSync(path)) {
+    fs.unlink(path, err => {
+      if (err) throw err;
```

Comment:
We should return a rejected promise, not throw an error which can't be handled.
Then, in `handleFileDelete`, we can do something like this:
```js
util.deleteFile(dest).catch(() => console.error("Unable to delete", dest));
```

Reply:
@nicolo-ribaudo Sorry about the delay, I didn't get any notification about your comment, I've now got them turned on. I've updated the PR based around your feedback, I hope this is what you were after.

I think that in the time between this PR being opened and now there's been another PR that looks like it's trying to achieve the same goal (https://github.com/babel/babel/pull/9303), so it looks like this is a feature that others are interested in. I know that in our situation it'd help eliminate some annoying issues.